addon/SearchProvider.js

```diff
@@ -41,6 +44,13 @@ SearchProvider.prototype = {
   ]),
 
   /**
+   * Event listener for the HIDDEN_ENGINES_PREF pref change
+   */
+  _onPrefChange(e) {
+    this.emit(SEARCH_ENGINE_TOPIC, "hiddenOneOffs");
```

Comment:
`SearchFeed` happens to just update its list of engines when it eventually gets the action broadcasted by `ActivityStreams`. But this is somewhat dangerous as the same action might have `data` of the string or the engine.